### PR TITLE
Unstiff falling

### PR DESCRIPTION
**Pull request by Vivituhh**
## Introduced Changes

To save roboter hardware the idea is to reduce the stiffnesses in falling shortly before the ground impact. There fore a new threshold is introduced for the angle after which the stiffness in the legs and arms is reduced to 20% (as Philip suggested in #356 ). To achieve that the stiffness is applied at the time of the impact, the `FallState` transition from `Falling` to `Fallen` is delayed such that the `FallProtection` can set the stiffness values.

## How to Test

Test with twix plots by looking at leg/arm stiffnesses which should be reduced at the `FallState`  transition from `Falling` to `Fallen` and when going to `StandUp` it should be increased again.

### CODE REVIEW

File: crates/types/src/configuration.rs [hunk 1]
```diff
@@ -487,10 +487,14 @@ pub struct CameraMatrixParameters {
 
 #[derive(Clone, Debug, Default, Deserialize, Serialize, SerializeHierarchy)]
 pub struct FallProtection {
+    pub ground_impact_angular_threshold: f32,
     pub ground_impact_head_stiffness: f32,
-    pub arm_stiffness: f32,
+    pub ground_impact_body_stiffness: f32,
+    pub time_free_motion_exit: f32,
```
Comment (julianschuler) [inline crates/types/src/configuration.rs:493]: Please use `Duration` instead of `f32` for time intervals.

**Review by julianschuler — CHANGES_REQUESTED** [review]
Nice addition! Still, I have a few remarks and have to test this out in the lab for myself.

File: crates/control/src/motion/fall_protector.rs [hunk 2]
```diff
@@ -3,32 +3,43 @@ use std::time::{Duration, SystemTime, UNIX_EPOCH};
...
 use context_attribute::context;
+use filtering::low_pass_filter::LowPassFilter;
 use framework::MainOutput;
 use motionfile::{MotionFile, MotionInterpolator};
+use nalgebra::Vector2;
 use types::{
-    configuration::FallProtection, BodyJoints, ConditionInput, CycleTime, FallDirection,
-    HeadJoints, Joints, JointsCommand, MotionCommand, MotionSelection, MotionType, SensorData,
+    configuration::FallProtection,
+    configuration::FallStateEstimation as FallStateEstimationConfiguration, BodyJoints,
```
[the remark this answers is not in the document]
Comment (Vivituhh) [reply crates/control/src/motion/fall_protector.rs]: Done

File: crates/control/src/motion/fall_protector.rs [hunk 3]
```diff
@@ -150,6 +201,25 @@ impl FallProtector {
...
 
+        self.last_fall_state = *context.fall_state;
+
+        match self.fallen_time {
+            Some(fallen_start)
+                if context
+                    .cycle_time
+                    .start_time
+                    .duration_since(fallen_start)
+                    .unwrap()
```
[the remark this answers is not in the document]
Comment (Vivituhh) [reply crates/control/src/motion/fall_protector.rs]: Done

File: crates/types/src/configuration.rs (hunk 1 above)
Comment (Vivituhh) [reply crates/types/src/configuration.rs]: Done

File: crates/control/src/motion/fall_protector.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (julianschuler) [reply crates/control/src/motion/fall_protector.rs]: Please do this for all `duration_since` calls.

File: crates/types/src/configuration.rs (hunk 1 above)
Comment (julianschuler) [reply crates/types/src/configuration.rs]: Please do this for all 'time' parameters.

**Review by julianschuler — APPROVED** [review]
LGTM!

**Pull request merged**